keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,531 @@
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+-->
+
+N/A
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+We embed a component config in the ConfigMap. The `apiVersion` field allows versioning.
+
+## Implementation History
+
+<!--
+Major milestones in the life cycle of a KEP should be tracked in this section.
+Major milestones might include
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Proposal` section being merged signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+This KEP is contingent on the maintainers of local clusters agreeing to it
+(specifically: Kind, K3d, Minikube, and Microk8s), since this is really about
+better documenting what they're already doing.
+
+This KEP is very narrow in scope and lightweight in implementation, and wouldn't
+make sense if there was a more ambitious local registry proposal.
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+At the sig-cluster-lifecycle meeting, we talked about many alternative
+proposals for where this data would be stored, including:
+
+- Using annotations on a Namespace
+- Using a plain ConfigMap
+- Using a Custom Resource
+
+With this proposal, I tried to strike a right balance between a format that
+followed existing conventions, and a format that didn't require too much
+Kubernetes API machinery to implement. The K3d team requested that we choose a
+solution that did not require any Kubernetes code dependencies.
```

Comment:
this is also a requirement for kind, fwiw. due to the nature of testing kubernetes, we strongly avoid depending on it.